tools/clang/lib/CodeGen/CGDecl.cpp

```diff
@@ -891,7 +891,7 @@ llvm::Value *CodeGenFunction::EmitLifetimeStart(uint64_t Size,
   llvm::CallInst *C =
       Builder.CreateCall(CGM.getLLVMLifetimeStartFn(), {SizeV, Addr});
   C->setDoesNotThrow();
-  if (CGM.getCodeGenOpts().HLSLEnablePartialLifetimeMarkers) return nullptr; // HLSL Change - Returning nullptr prevents generating lifetime.end
+  if (CGM.getCodeGenOpts().HLSLExtensionsCodegen && !CGM.getCodeGenOpts().HLSLExtensionsCodegen->IsOptionEnabled(hlsl::options::TOGGLE_PARTIAL_LIFETIME_MARKERS)) return nullptr; // HLSL Change - Returning nullptr prevents generating lifetime.end
```

Comment:
I think it would be nice to have a helper function that can do the check in a single call instead of having to always check for nullptr.

So we could write something like 

`if (CGM.getCodeenOpts().IsFooEnabled()) {...}` 

instead of 

`if (GM.getCodeGenOpts().HLSLExtensionsCodegen && !CGM.getCodeGenOpts().HLSLExtensionsCodegen->IsOptionEnabled(hlsl::options::TOGGLE_FOO)`